packages/components/src/tabs/stories/index.story.tsx

```diff
@@ -0,0 +1,316 @@
+/**
+ * External dependencies
+ */
+import type { Meta, StoryFn } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { wordpress, more, link } from '@wordpress/icons';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Tabs from '..';
+import Popover from '../../popover';
+import { Slot, Fill, Provider as SlotFillProvider } from '../../slot-fill';
+import DropdownMenu from '../../dropdown-menu';
+import Button from '../../button';
+
+const meta: Meta< typeof Tabs > = {
+	title: 'Components/Tabs',
+	component: Tabs,
+	parameters: {
+		actions: { argTypesRegex: '^on.*' },
+		controls: { expanded: true },
+		docs: { canvas: { sourceState: 'shown' } },
+	},
+};
+export default meta;
+
+const Template: StoryFn< typeof Tabs > = ( props ) => {
+	return (
+		<Tabs { ...props }>
+			<Tabs.TabList>
+				<Tabs.Tab id={ 'tab1' } title={ 'Tab 1' }>
+					Tab 1
+				</Tabs.Tab>
+				<Tabs.Tab id={ 'tab2' } title={ 'Tab 2' }>
+					Tab 2
+				</Tabs.Tab>
+				<Tabs.Tab id={ 'tab3' } title={ 'Tab 3' }>
+					Tab 3
+				</Tabs.Tab>
+			</Tabs.TabList>
+			<Tabs.TabPanel id={ 'tab1' }>
+				<p>Selected tab: Tab 1</p>
+			</Tabs.TabPanel>
+			<Tabs.TabPanel id={ 'tab2' }>
+				<p>Selected tab: Tab 2</p>
+			</Tabs.TabPanel>
+			<Tabs.TabPanel id={ 'tab3' }>
+				<p>Selected tab: Tab 3</p>
+			</Tabs.TabPanel>
+		</Tabs>
+	);
+};
+
+export const Default = Template.bind( {} );
+
+const DisabledTabTemplate: StoryFn< typeof Tabs > = ( props ) => {
+	return (
+		<Tabs { ...props }>
+			<Tabs.TabList>
+				<Tabs.Tab id={ 'tab1' } title={ 'Tab 1' } disabled={ true }>
+					Tab 1
+				</Tabs.Tab>
+				<Tabs.Tab id={ 'tab2' } title={ 'Tab 2' }>
+					Tab 2
+				</Tabs.Tab>
+				<Tabs.Tab id={ 'tab3' } title={ 'Tab 3' }>
+					Tab 3
+				</Tabs.Tab>
+			</Tabs.TabList>
+			<Tabs.TabPanel id={ 'tab1' }>
+				<p>Selected tab: Tab 1</p>
+			</Tabs.TabPanel>
+			<Tabs.TabPanel id={ 'tab2' }>
+				<p>Selected tab: Tab 2</p>
+			</Tabs.TabPanel>
+			<Tabs.TabPanel id={ 'tab3' }>
+				<p>Selected tab: Tab 3</p>
+			</Tabs.TabPanel>
+		</Tabs>
+	);
+};
+export const DisabledTab = DisabledTabTemplate.bind( {} );
+
+const WithTabIconsAndTooltipsTemplate: StoryFn< typeof Tabs > = ( props ) => {
+	return (
+		// SlotFill is used here to ensure the icon's tooltips are not
+		// rendered inline, as that would cause them to inherit the tab's opacity.
+		<SlotFillProvider>
```

Comment:
I'm sure you are aware after helping review the [Tooltip refactor](https://github.com/WordPress/gutenberg/pull/48440/files#diff-3cd08a9b01995f5596d283ce970bfc6ac006847cbe717d6098e680ffd85ece28), but just wanted to add a reminder here about the updated tests in `TabPanel`: which involved removing `<SlotFillProvider>` and adding the `cleanupTooltip()` utility function in tests where a `tooltip` is rendered.

Reply:
Thank you for the reminder! 8fb8b7e48b